packages/vue/src/components/SpAssets/SpAssets.vue

```diff
@@ -22,31 +48,125 @@
             {{ new Intl.NumberFormat('en-GB').format(balance.amount) }}
           </td>
         </tr>
+        <tr v-if='filteredBalanceList<=0 && searchQuery !== ""' class='assets-table__row'>
+          <td class='assets-table__row--no-results' colspan='2'>
+            <h4>No results for '{{ searchQuery }}' </h4>
+            <p>Try again with another search</p>
+          </td>
+        </tr>
       </tbody>
     </table>
+    <template v-if='uiState.isAssetsLoading'>
+      <div v-for='n in 2' :key='n' class='loading__row'>
+        <div class='loading__col'>
+          <span class='loading__avatar'></span>
+          <span class='loading__denom'></span>
+        </div>
+        <div class='loading__col loading__col--justify-end'>
+          <span class='loading__ibc'></span>
+        </div>
+        <div class='loading__col loading__col--justify-end'>
+          <span class='loading__amount'></span>
+        </div>
+      </div>
+    </template>
+    <div v-if='!uiState.isAssetsLoading && !balances.length' class='no-result-label'>You have no assets</div>
+    <div v-if='!uiState.isAssetsLoading && isShowMore' class='show-more' @click='showMore()'>
+      Show more
+    </div>
   </section>
 </template>
 
 <script lang="ts">
-import { computed, defineComponent, onMounted, PropType } from 'vue'
+import { computed, defineComponent, onMounted, PropType, reactive, ref, toRefs, watch } from 'vue'
 import { useStore } from 'vuex'
 
+import SpDenom from '../SpDenom'
+
+export interface State {
+  isAssetsLoading: boolean
+}
+
+export let initialState: State = {
+  isAssetsLoading: true
+}
+
 export default defineComponent({
   name: 'SpAssets',
+  components: { SpDenom },
 
   props: {
     address: {
       type: String as PropType<string>
+    },
+    resultLimit: {
+      type: Number,
+      default: 3,
+      required: false
     }
   },
 
   setup(props) {
     // store
     const $s = useStore()
+    const searchInput = ref<null | { focus: () => null }>(null)
```

Comment:
I think we can go for a single state object

Reply:
This is how it was intended, TS complains about focus(). I didn't find any other solution to solve this problem other than to create another ref value.